src/goto-analyzer/goto_analyzer_parse_options.cpp

```diff
@@ -131,6 +139,232 @@ void goto_analyzer_parse_optionst::get_command_line_options(optionst &options)
+  {
+    options.set_option("verify", true);
+    options.set_option("general-analysis", true);
+  }
+  else if(cmdline.isset("simplify"))
+  {
+    options.set_option("simplify", true);
+    options.set_option("outfile", cmdline.get_value("simplify"));
+    options.set_option("general-analysis", true);
+
+    // For development allow slicing to be disabled in the simplify task
+    options.set_option(
+      "simplify-slicing",
+      !(cmdline.isset("no-simplify-slicing")));
+  }
+  else if(cmdline.isset("show-intervals"))
+  {
+    // For backwards compatibility
+    options.set_option("show", true);
+    options.set_option("general-analysis", true);
+    options.set_option("intervals", true);
+    options.set_option("domain set", true);
+  }
+  else if(cmdline.isset("(show-non-null)"))
+  {
+    // For backwards compatibility
+    options.set_option("show", true);
+    options.set_option("general-analysis", true);
+    options.set_option("non-null", true);
+    options.set_option("domain set", true);
+  }
+  else if(cmdline.isset("intervals") || cmdline.isset("non-null"))
+  {
+    // For backwards compatibility either of these on their own means show
+    options.set_option("show", true);
+    options.set_option("general-analysis", true);
+  }
+
+  if(options.get_bool_option("general-analysis") || reachability_task)
+  {
+    // Abstract interpreter choice
+    if(cmdline.isset("location-sensitive"))
+      options.set_option("location-sensitive", true);
+    else if(cmdline.isset("concurrent"))
+      options.set_option("concurrent", true);
+    else
+    {
+      // Silently default to location-sensitive as it's the "default"
+      // view of abstract interpretation.
+      options.set_option("location-sensitive", true);
+    }
+
+    // Domain choice
+    if(cmdline.isset("constants"))
+    {
+      options.set_option("constants", true);
+      options.set_option("domain set", true);
+    }
+    else if(cmdline.isset("dependence-graph"))
+    {
+      options.set_option("dependence-graph", true);
+      options.set_option("domain set", true);
+    }
+    else if(cmdline.isset("intervals"))
+    {
+      options.set_option("intervals", true);
+      options.set_option("domain set", true);
+    }
+    else if(cmdline.isset("non-null"))
+    {
+      options.set_option("non-null", true);
+      options.set_option("domain set", true);
+    }
+
+    // Reachability questions, when given with a domain swap from specific
+    // to general tasks so that they can use the domain & parameterisations.
+    if(reachability_task)
+    {
+      if(options.get_bool_option("domain set"))
+      {
+        options.set_option("specific-analysis", false);
+        options.set_option("general-analysis", true);
+      }
+    }
+    else
+    {
+      if(!options.get_bool_option("domain set"))
+      {
+        // Default to constants as it is light-weight but useful
+        status() << "Domain defaults to --constants" << eom;
```

Comment:
Is it worth tweaking this status message to say something like 'Domain not specified, defaulting to --constants' ? I just wonder if thats a clearer statement of the situation.

Reply:
Fixed.